stm32-modules/include/heater-shaker/heater-shaker/heater_task.hpp

```diff
@@ -451,6 +453,14 @@ class HeaterTask {
     auto visit_conversion(double value, TemperatureSensor& sensor) -> void {
         if (value > sensor.overtemp_limit_c) {
             sensor.error = sensor.overtemp_error;
+        } else if ((value > (sensor.overtemp_limit_c -
```

Comment:
A comment in this function documenting the error logic would be nice.

Reply:
agree, and it might even be a good idea to factor the conditionals into their own `static inline` functions. That way, you can call the functions in the if statement, but inside the functions you're not constraints to having a single expression, so you could do like

```c++
static inline bool is_hardware_overtemp(double value, TemperatureSensor& sensor, uint8_t bitmap) {
    if (value > (sensor.overtemp_limit_c - HEATER_PAD_HARDWARE_OVERTEMP_OFFSET_C) ) {
       if ((bitmap & State::POWER_GOOD_ERROR) != 0) {
          return true;
       } 
    }
    return false;
}
```

Reply:
i guess it depends whether you find that more readable or not; I thought it was pretty strictly an improvement until I typed it out and now I think it's kind of up to preference.